pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -0,0 +1,108 @@
+package gkenetworkparamset
+
+import (
+	"context"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
+	"github.com/onsi/gomega"
+	"google.golang.org/api/compute/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cloud-provider-gcp/crd/apis/network/v1alpha1"
+	"k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned/fake"
+	networkinformers "k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions"
+	"k8s.io/cloud-provider-gcp/providers/gce"
+	"k8s.io/component-base/metrics/prometheus/controllers"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+func TestControllerRuns(t *testing.T) {
+	ctx, stop := context.WithCancel(context.Background())
+
+	fakeNetworking := fake.NewSimpleClientset()
+	networkInformerFactory := networkinformers.NewSharedInformerFactory(fakeNetworking, controller.NoResyncPeriodFunc())
+
+	fakeGCE := gce.NewFakeGCECloud(gce.DefaultTestClusterValues())
+	controller := NewGKENetworkParamSetController(
+		fakeNetworking,
+		networkInformerFactory.Networking().V1alpha1().GKENetworkParamSets(),
+		networkInformerFactory.Start,
+		fakeGCE,
+	)
+	metrics := controllers.NewControllerManagerMetrics("test")
+	go controller.Run(ctx.Done(), metrics)
+	stop()
+}
+
+func TestAddValidParamSet(t *testing.T) {
```

Comment:
Would be good to add more test cases.
1. More than one range on the subnet/gkenetworkparamset
2. No match to the pod range
3. Only a partial match to the ranges.

Reply:
done. Added in test cases for subnet match too.